src/libstd/io/buffered.rs

```diff
@@ -30,7 +31,9 @@ use ptr;
 #[stable(feature = "rust1", since = "1.0.0")]
 pub struct BufReader<R> {
     inner: R,
-    buf: Cursor<Vec<u8>>,
+    buf: Vec<u8>,
+    pos: usize,
+    cap: usize,
```

Comment:
Yeah I tried to think of ways to keep it as a `Cursor` but nothing really seemed right, especially since `Cursor<Vec<u8>>` extends the vec on its own.